compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
If size_t were to change to an integer of size 128 tomorrow, a program that depended on unsigned ints overflowing at a certain point would break.  A program adds 1 to a size_t would not.  One of these actions depends on the underlying type definition.  The other depends on a property of the type that is necessary for its functionality.  The former case is leaked through telling the default definition, the latter should be presented by the documenter for someone to use the type.  You can get all the latter information through merely having an understanding of what the type *does*, while the former information is only available through knowing what the type *is*.